### ISSUE TITLE
Make Error

### ISSUE THREAD

**Issue by nickman602** [nicehash/nheqminer#1]
collect2: error: ld returned 1 exit status
make: *** [nheqminer] Error 1

Running Ubuntu 14.04.1 LTS 

I already had the wallet installed, and had done some cpu mining with the other standalone miner.  decided to try this out and installed Qt5 as directed beforehand. This is all being done in command line over ssh.

**Comment by S74nk0** [comment]
Hi @nickman602. What qmake and gcc versions are installed on the system?

**Comment by nickman602** [comment]
Hi @S74nk0 . Thanks for helping me. Below are the two outputs when I check for versions.

gcc version 4.8.4 (Ubuntu 4.8.4-2ubuntu1~14.04.3)

QMake version 3.0
Using Qt version 5.2.1 in /usr/lib/x86_64-linux-gnu

**Comment by S74nk0** [comment]
Hi @nickman602. I will test this on Ubuntu 14.04.3 and report back what could be the problem. We will most likely be making updates to the build system for Linux to make this process more streamlined.

**Comment by S74nk0** [comment]
Also what is the CPU on the system?

**Comment by nickman602** [comment]
Thanks, and sorry for the trouble.

It's an Intel(R) Xeon(R) X7560. Too old?

**Comment by pattkung** [comment]
same error on ubuntu 16.04 too. I try to run on vps

**Comment by danwxyz** [comment]
Same error. 14.04

```
/builds/nheqminer/nheqminer/libs/linux_ubuntu/libboost_thread.a(thread.o): In function `void std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >::_M_initialize_dispatch<boost::iterators::transform_iterator<boost::algorithm::detail::copy_iterator_rangeF<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, __gnu_cxx::__normal_iterator<char*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >, boost::algorithm::split_iterator<__gnu_cxx::__normal_iterator<char*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >, boost::iterators::use_default, boost::iterators::use_default> >(boost::iterators::transform_iterator<boost::algorithm::detail::copy_iterator_rangeF<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, __gnu_cxx::__normal_iterator<char*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >, boost::algorithm::split_iterator<__gnu_cxx::__normal_iterator<char*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >, boost::iterators::use_default, boost::iterators::use_default>, boost::iterators::transform_iterator<boost::algorithm::detail::copy_iterator_rangeF<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, __gnu_cxx::__normal_iterator<char*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >, boost::algorithm::split_iterator<__gnu_cxx::__normal_iterator<char*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >, boost::iterators::use_default, boost::iterators::use_default>, std::__false_type)':
thread.cpp:(.text._ZNSt6vectorINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESaIS5_EE22_M_initialize_dispatchIN5boost9iterators18transform_iteratorINS9_9algorithm6detail20copy_iterator_rangeFIS5_N9__gnu_cxx17__normal_iteratorIPcS5_EEEENSC_14split_iteratorISI_EENSA_11use_defaultESM_EEEEvT_SO_St12__false_type[_ZNSt6vectorINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESaIS5_EE22_M_initialize_dispatchIN5boost9iterators18transform_iteratorINS9_9algorithm6detail20copy_iterator_rangeFIS5_N9__gnu_cxx17__normal_iteratorIPcS5_EEEENSC_14split_iteratorISI_EENSA_11use_defaultESM_EEEEvT_SO_St12__false_type]+0x238): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned long&, unsigned long)'
collect2: error: ld returned 1 exit status
make: *** [nheqminer] Error 1

```


```
$ gcc --version
gcc (Ubuntu 4.9.4-2ubuntu1~14.04.1) 4.9.4
```


```
$ qmake --version
QMake version 3.0
Using Qt version 5.2.1 in /usr/lib/x86_64-linux-gnu
```

**Comment by asluchevskiy** [comment]
Hi, confirming the same error on 14.04 as above.

**Comment by nickman602** [comment]
Tried on a fresh install of Debian 8.

Got a different error:

collect2: error: ld returned 1 exit status
Makefile:177: recipe for target 'nheqminer' failed
make: *** [nheqminer] Error 1

gcc (Debian 4.9.2-10) 4.9.2

QMake version 3.0
Using Qt version 5.3.2 in /usr/lib/x86_64-linux-gnu

**Comment by Ales999** [comment]
Success make and Work:
~$ cat /etc/os-release 
NAME="Ubuntu"
VERSION="16.04.1 LTS (Xenial Xerus)"

**Comment by JanKalin** [comment]
Solved on `Debian GNU/Linux 7.11 (wheezy)` with `gcc (Debian 4.7.2-5) 4.7.2`. There's a library missing from the linker line.

Do

- `echo "LIBS += -lrt" >> ../nheqminer/nheqminer.pro`

before running 

- `qmake ../nheqminer/nheqminer.pro`

which will add the missing library.

**Comment by S74nk0** [comment]
Hi all! We have added a new way to build binaries on Linux with CMake (recommended). This has been tested on Ubuntu 14.04 and 16.04 Desktop and Ubuntu 14.04 Server.

Follow [these](https://github.com/nicehash/nheqminer#linux-cmake-recommended-tested-on-ubuntu-desktop-1404-and-1604-and-ubuntu-server-1404) instructions. We will update how to build for for other distros as well.
